rcljava/src/test/java/org/ros2/rcljava/node/NodeParametersTest.java

```diff
@@ -0,0 +1,383 @@
+/* Copyright 2020 Open Source Robotics Foundation, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ros2.rcljava.node;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.AfterClass;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.ros2.rcljava.RCLJava;
+import org.ros2.rcljava.parameters.ParameterCallback;
+import org.ros2.rcljava.parameters.ParameterType;
+import org.ros2.rcljava.parameters.ParameterVariant;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+public class NodeParametersTest {
+  private Node node;
+  private boolean callbackCalled = false;
+
+  @BeforeClass
+  public static void setupOnce() throws Exception {
+    // Just to quiet down warnings
+    org.apache.log4j.BasicConfigurator.configure();
+
+    RCLJava.rclJavaInit();
+  }
+
+  @AfterClass
+  public static void tearDownOnce() {
+    RCLJava.shutdown();
+  }
+
+  @Before
+  public void setUp() {
+    node = RCLJava.createNode("test_node");
+  }
+
+  @After
+  public void tearDown() {
+    node.dispose();
+  }
+
+  @Test
+  public final void testDeclareParameter() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
We don't have to do it as part of this PR, but we should consider adding custom exceptions for parameter failure cases.
If you agree, I can make a follow-up ticket to track it.

Reply:
Yeah, completely agreed.  Custom exceptions would make this much nicer for the user.  I also agree with making a follow-up ticket, since I don't want to make this already large PR even larger.  I'll go ahead and do that.

Reply:
Oh, except we can't because this is a fork.  Not sure how to proceed here.

Reply:
I'll make a note locally for after this is upstreamed. We could also add a TODO in the code somewhere.

Reply:
Good call about the TODO.  I've added one now.